core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -394,6 +395,12 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
             return;
         }
 
+        if (evt instanceof DisconnectWhenFinished) {
+            assert responseDecoder != null;
+            responseDecoder.disconnectWhenFinished();
+            return;
+        }
```

Comment:
I think we can access `HttpSessionHandler` from `ctx` via `HttpSession.get(Channel)`. We could add `HttpSession.disconnectWhenFinished()` so that `HttpSessionHandler` calls `responseDecoder.disconnectWhenFinished()` for us.

Reply:
That's good idea.
But I can't access `HttpSession.get(Channel)` in `DefaultClientRequestContext`

Reply:
Let's move `HttpSession` into `internal.client` and make it public.